bumpCversion.py

```diff
@@ -46,20 +20,55 @@ def modify_revision(matchobj, action):
     and name.
     """
     currentVer = int(matchobj.group('val'))
-
+    retStr = ''
+  
     if (action == 'zero'):
+        change_bump_on_zero()
         newVer = 0
     elif (action == 'bump'):
         newVer = currentVer + 1
     else:
         raise ValueError("Invalid 'action' parameter " + str(action))
 
-    retStr = "#define " + matchobj.group('varName') + matchobj.group('nSpaces') + \
+
+    try:
+        retStr = "#define " + matchobj.group('varName') + matchobj.group('nSpaces') + \
         "(" + str(newVer) + matchobj.group('unsigned') + ")"
+        return (str(retStr))
+    except IndexError:
+        print("WARNING: " + str(DoxyException(matchobj)))
+        retStr = get_doxy_str(matchobj, newVer)
+        pass
```

Comment:
This is an interesting approach. I think it would be better to pull this into to seperate functions: one for C and one for Doxyfiles.